akka-actor/src/main/java/akka/dispatch/affinity/OnSpinWait.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.dispatch.affinity;
+
+import akka.util.Unsafe;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import scala.util.control.NonFatal;
+import static java.lang.invoke.MethodType.methodType;
+
+final class OnSpinWait {
+    private final static MethodHandle handle;
+
+    public final static void spinWait() throws Throwable {
+        handle.invokeExact();
+    }
+
+    static {
+        final MethodHandle noop = MethodHandles.constant(Object.class, null).asType(methodType(Void.TYPE));
+        MethodHandle impl;
+        try {
+          impl = MethodHandles.lookup().findStatic(Thread.class, "onSpinWait", methodType(Void.TYPE));
+        } catch (NoSuchMethodException nsme) {
+          impl = noop;
+        } catch (IllegalAccessException iae) {
+          impl = noop;
+        }
+        handle = impl;
+  };
```

Comment:
That's great that we can start using that and fall back to noop for JDK8. There are more places we can make use of this so it can live in package `akka.util` or `akka.dispatch`.

Reply:
True—I'm still wrestling with `native-image` to get it to inline appropriately so I'll revisit once I have a configuration which works. Will then update the PR to move it to akka.util for internal usage.

Reply:
Did you make native image happy with this?

Reply:
Yes and no, not OOTB, but I was then able to add a native-image Substitution-rule which calls Thread.onSpinWait() directly. So I guess you could say that it solved the problem. :S